Use clang toolset by default as we have no gcc in main.

--- a/tools/build/src/build-system.jam
+++ b/tools/build/src/build-system.jam
@@ -633,7 +633,7 @@ local rule should-clean-project ( project )
         }
         else
         {
-            default-toolset = gcc ;
+            default-toolset = clang ;
             if [ os.name ] = NT
             {
                 default-toolset = msvc ;
--- a/tools/build/test/BoostBuild.py
+++ b/tools/build/test/BoostBuild.py
@@ -108,7 +108,7 @@ def get_toolset():
     if sys.platform == "darwin" or sys.platform.startswith("freebsd") or sys.platform.startswith("openbsd"):
         return "clang"
 
-    return "gcc"
+    return "clang"
 
 
 # Detect the host OS.
@@ -349,7 +349,7 @@ class Tester(TestCmd.TestCmd):
             pass
 
     def set_toolset(self, toolset, target_os=None, _pass_toolset=True):
-        self.toolset = _pass_toolset and toolset or "gcc"
+        self.toolset = _pass_toolset and toolset or "clang"
         if not target_os and self.toolset.startswith("emscripten"):
             target_os = "none"
         self.target_os = target_os or host_os
